### PR TITLE
create store with redux-toolkit (rtk)

### PR DESCRIPTION
**Pull request by johnmarkredding**
Switched from redux `createStore` to rtk `configureStore`.
Exported custom store type (`AppStore`) to include strong typing for the specifics of our state.

I want to have this pulled to see if we're on the right track, and this will play nicely before moving on. 

The next steps I see are using `slices` instead of actions/reducers explicitly, but that will take more time as there are some expections to the normal action and reducer patterns I see.
Ex: 'cancelSubscriptions' doesn't belong to a reducer in the way that the other actions do.
Also the functions `computeForecasts` and `forecasts` in the reducer for `sounding.ts` don't fit super nicely into the slice pattern, but we can make that work fine, if we want to go with slices.

### CODE REVIEW

**Comment by vicb** [comment]
Thank you so much @johnmarkredding !
I'll review the PR tomorrow CET

File: src/util/store.ts
Comment (vicb) [inline src/util/store.ts]: this should not be needed because it's the default :)

see https://redux-toolkit.js.org/api/getDefaultMiddleware#intended-usage

May be add:

`devTools: process.env.NODE_ENV !== 'production',`

as in devTools: process.env.NODE_ENV !== 'production',

**Review by vicb — CHANGES_REQUESTED** [review]
Looks good.

I prefer to have one commit per purpose, i.e.
- 1 commit to fix the typos/formatting (thank you for that)
- 1 commit for the code change (adding RTK + usage)

Don't worry if you're not sure about how to do. I can help you / do it / keep everything in a single and we will not the tools stand in our way

File: src/util/store.ts
Comment (johnmarkredding) [reply src/util/store.ts]: Oh, yes. I didn't mean to leave that included. Sorry! Let me fix that

Comment (johnmarkredding) [reply src/util/store.ts]: As for the `devTools`, I thought that was handled by the toolkit. Will fix it.

Comment (johnmarkredding) [reply src/util/store.ts]: I squashed the commits, if that is what you were wanting. I hope this looks better.

**Comment by vicb** [comment]
Thanks @johnmarkredding, the PR looks great.

I'll merge it next week when I'm back home - do not worry about the conflicting `npm-shrinkwrap.json`, I'll re-generate this one after the commits are merged.

**Comment by vicb** [comment]
I have just merged the PR, thanks!

**Pull request closed without merging**
